internal/middlewares/require_auth.go

```diff
@@ -0,0 +1,123 @@
+package middlewares
+
+import (
+	"github.com/valyala/fasthttp"
+
+	"github.com/authelia/authelia/v4/internal/authentication"
+	"github.com/authelia/authelia/v4/internal/model"
+	"github.com/authelia/authelia/v4/internal/session"
+)
+
+// Require1FA check if user has enough permissions to execute the next handler.
+func Require1FA(next RequestHandler) RequestHandler {
+	return func(ctx *AutheliaCtx) {
+		if s, err := ctx.GetSession(); err != nil || s.AuthenticationLevel < authentication.OneFactor {
+			ctx.ReplyForbidden()
+			return
+		}
+
+		next(ctx)
+	}
+}
+
+// RequireElevated requires various elevation criteria.
+//
+//nolint:gocyclo
+func RequireElevated(next RequestHandler) RequestHandler {
```

Comment:
Could we split this to each elevation level for functional separation and readability/complexity?